modules/contact/src/actions/ContactAction.C

```diff
@@ -532,12 +541,20 @@ ContactAction::addMortarContact()
       const auto primal_type =
           _problem->systemBaseNonlinear().system().variable_type(displacements[0]);
 
-      const int lm_order = primal_type.order.get_order();
-
       if (primal_type.family == LAGRANGE)
       {
         params.set<MooseEnum>("family") = Utility::enum_to_string<FEFamily>(primal_type.family);
-        params.set<MooseEnum>("order") = Utility::enum_to_string<Order>(OrderWrapper{lm_order});
+        if (_lm_order == 0) // PRIMAL
+        {
+          const int lm_order = primal_type.order.get_order();
+          params.set<MooseEnum>("order") = Utility::enum_to_string<Order>(OrderWrapper{lm_order});
+        }
+        else if (_lm_order == 1) // FIRST
+          params.set<MooseEnum>("order") = "FIRST";
+        else if (_lm_order == 2) // SECOND
```

Comment:
For dual mortar, which is our recommended mortar mechanical contact approach because, among other things, yields a verifiable evolution of contact forces (Lagrange multipliers), the LM interpolation order is the same as the primal interpolation order. We did some good testing without dual bases and choosing lower orders for the LMs, but never got rid of pressure instabilities. For those reasons, I am not sure that we should have these options in the action. If a user wants to build the objects in the input file and select the different interpolation orders, that is fine; but, I'd be hesitant to provide those options through the action unless there is a practical reason.

Reply:
Okay, that's new to me. I was using First LM with second displacements with dual basis. I did see some differences in testing, but I can check it against my big simulations to see if there is any big problem.

Reply:
We made comparisons with analytical contact pressure solution of "simple" benchmark problems, using cylinder-on-plane problems --should be in FY21 report. While you can get the problem to converge, the LM solution is not fully stable; which is seen, in practice, with a non-smooth evolution of the interpolated LM field in many cases. All these depend on geometry and level of mesh refinement.

It is true that we have the option not to do dual interpolation in the action. But I think the user of the action would want to always do dual, or 99.9% of the times.

Reply:
This will require update and (I think) a regolding of some of the non-action tests in the combined module.

Reply:
Those tests (tinkering with non dual interpolation order) could stay as they are without their action counterparts. Unless you want to port them.

Reply:
Sure...kinda adds to the confusion though on what parameters to use? Any reason to keep them?

Reply:
No reason on my side. I felt more strongly about the action because it could mislead users that are not experienced with contact. But, yeah, making those tests consistent with the recommended settings is great.

Reply:
Yeah, diff's are mostly like this:
```
test:gap_heat_transfer_mortar/finite-2d-rz.thermomechanical_mortar_action/second_high_k_action: In file second_high_k_action.csv: The values in column "contact" don't match @ t3
test:gap_heat_transfer_mortar/finite-2d-rz.thermomechanical_mortar_action/second_high_k_action: 	relative diff:   5.000e+00 ~ 9.000e+00 = 4.444e-01 (4.444e-01)
```

Reply:
No scratch that, the diffs on the non-action are just due to 1e-5 difference in temperature. the action has those diffs above which indicate something else is screwy...